src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c

```diff
@@ -492,7 +492,7 @@ int32_t GlobalizationNative_IndexOf(
         result = GlobalizationNative_CompareString(pSortHandle, lpTarget, cwTargetLength, lpSource, cwSourceLength, options);
         if (result == UCOL_EQUAL && pMatchedLength != NULL)
         {
-            *pMatchedLength = cwTargetLength;
+            *pMatchedLength = cwSourceLength;
```

Comment:
This was incorrectly returning the target length instead of the source length and was causing some newly introduced unit tests to fail.

Reply:
Don't know if it requires backporting, to be honest. Here's a simplified repro:

```cs
var newString = "".Replace("\u200d", "x", StringComparison.InvariantCulture);
```

Before this fix, this throws an `ArgumentOutOfRangeException` (due to attempting to slice a span in an invalid fashion). After the fix, this correctly returns `""`.

To trigger the bug, the "this" string must be empty, _and_ the "old" string must be non-empty and consist only of zero-weight chars, _and_ the caller must be passing a linguistic comparison such as `InvariantCulture`. I think the likelihood of all three happening at the same time are low enough where we couldn't justify the backport.

Reply:
I agree we don't have to service this till we get a case need it.